benchmarks/htap/lib/stats.py

```diff
@@ -194,16 +202,42 @@ def oltp_counts(self, query_type = None):
         return (ok, err)
 
     def oltp_total(self, query_type = None):
-        data = self.data['oltp']
-        # If query_type is None, then flatten all records into a single list, otherwise just take the records of the
-        # requested type
-        stats = data[query_type] if query_type != None else [item for sublist in data.values() for item in sublist]
-        ok, err = [], []
-        for s in stats:
-            (ok if s['status'] == 'ok'  else err).append(s)
-        tps = self._oltp_tps(ok)
-        latency = self._oltp_latency(ok)
-        return (tps, latency)
+        oltp = self.data['oltp']
+        # If query_type is None, then flatten all records into a single list, otherwise just take the records of the requested type
+        tps = []
+        total_txs = 0
+        total_acc_runtime = 0
+        min_runtime = float('inf')
+        max_runtime = 0
+        avg_latency_last_bucket = 0
+        # TODO ignore first and last 10% in the history
+        for bucket in oltp['history']:
+            bucket_stats = [bucket[1][query_type]] if query_type != None else bucket[1].values()
+
+            bucket_ok_txs = 0
+            bucket_total_txs = 0
+            bucket_total_acc_runtime = 0
+            for s in bucket_stats:
+                # For TPS:
+                bucket_ok_txs += s.get_ok_transactions()
+
+                # For latencies
+                runtimes = s.get_runtimes()
+                min_runtime = min(min_runtime, runtimes[0])
+                max_runtime = max(max_runtime, runtimes[1])
+                total_acc_runtime += runtimes[2]
+                bucket_total_acc_runtime += runtimes[2]
+                num_txs = s.get_total_transactions()
+                total_txs += num_txs
+                bucket_total_txs += num_txs
+
+            tps.append(bucket_ok_txs)
+            last_avg_latency = int(bucket_total_acc_runtime/bucket_total_txs) if bucket_total_txs != 0 else 0
```

Comment:
you can also keep an array like tps and then take the -1 index on that one and have a complete result for sure?

Reply:
yes, I could, but why is this not complete?
For latency we are not interested in latency per second or anything like that at the moment, just latency across the whole history range. Except for the last second as we print the CURR value (which I'm not sure if it is very valuable).

Reply:
i mean for the value of `last_avg_latency` as this is now taken from an incomplete bucket

Reply:
changed to output tps[-2] and latency[-2], so not the current second but the previous one.